### PR TITLE
Allow using all auth types by default, and also making authentication optional

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

File: plugins/authentication/RestfulAuthenticationManager.php [hunk 1]
```diff
@@ -14,6 +14,37 @@ class RestfulAuthenticationManager extends \ArrayObject {
...
+   *
+   * @param boolean $is_optional
+   *   Determines if the authentication is optional.
+   */
+  public function setIsOptional($is_optional) {
+    $this->isOptional = $is_optional;
+  }
+
+  /**
+   * Set the authentications' "optional" flag.
```
Comment (snirs90) [inline plugins/authentication/RestfulAuthenticationManager.php]: Set => Get

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch in case you missed the PR ^^ :)

File: restful.module [hunk 2]
```diff
@@ -263,11 +269,23 @@ function restful_get_restful_handler_by_name($plugin_name) {
...
-  foreach ($plugin['authentication_types'] as $auth_plugin_name) {
+  foreach ($auth_plugins as $auth_plugin_name) {
     $auth_handler = restful_get_authentication_handler($auth_plugin_name);
     $handler->getAuthenticationManager()->addAuthenticationProvider($auth_handler);
+    $has_auth = TRUE;
   }
+
+  if ($has_auth) {
+    // Set the "optional" flag of the authentication manager.
+    $handler->getAuthenticationManager()->setIsOptional($plugin['authentication_optional']);
```
Comment (e0ipso) [inline restful.module]: Can we refactor all the `$has_auth` logic into:

``` php
if (!empty($auth_plugins)) {
  // Set the "optional" flag of the authentication manager.
  $handler->getAuthenticationManager()->setIsOptional($plugin['authentication_optional']);
}
```

But I don't get why we need the check, couldn't we just set it anyways? Maybe even by the authentication manager constructor and avoid the extra logic?

**Comment by e0ipso** [comment]
I was missing this feature just today! One minor detail, other than that this is a go.

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**